patches/graphql-relay+0.5.3.patch

```diff
@@ -0,0 +1,30 @@
+patch-package
+--- a/node_modules/graphql-relay/lib/connection/arrayconnection.js
++++ b/node_modules/graphql-relay/lib/connection/arrayconnection.js
+@@ -61,7 +61,8 @@ function connectionFromArraySlice(arraySlice, args, meta) {
+       first = args.first,
+       last = args.last;
+   var sliceStart = meta.sliceStart,
+-      arrayLength = meta.arrayLength;
++      arrayLength = meta.arrayLength,
++      nodeField= meta.nodeField;
+ 
+   var sliceEnd = sliceStart + arraySlice.length;
+   var beforeOffset = getOffsetWithDefault(before, arrayLength);
+@@ -88,10 +89,14 @@ function connectionFromArraySlice(arraySlice, args, meta) {
+   var slice = arraySlice.slice(Math.max(startOffset - sliceStart, 0), arraySlice.length - (sliceEnd - endOffset));
+ 
+   var edges = slice.map(function (value, index) {
+-    return {
++    const newEdge = {
+       cursor: offsetToCursor(startOffset + index),
+-      node: value
++      node: !!nodeField ? value[nodeField] : value,
+     };
++    if (nodeField) {
++      return Object.assign({ ...value }, newEdge, {});
```

Comment:
Btw, if you’re using spread already anyways, this is equivalent: `{ ...value, ...newEdge }`